staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -312,9 +347,16 @@ func (s *APIAggregator) AddAPIService(apiService *v1.APIService) error {
 	}
 	proxyHandler.updateAPIService(apiService)
 	if s.openAPIAggregationController != nil {
-		s.openAPIAggregationController.AddAPIService(proxyHandler, apiService)
+		// this is calling a controller.  It should already handle being async.
```

Comment:
@deads2k what was the reason for these two changes?

Reply:
@marun alternative: https://github.com/kubernetes/kubernetes/pull/89147

Reply:
@sttts By alternative, do you mean I should drop this commit in favor of that PR immediately? I don't have criteria for choosing.

Reply:
yes. for the upper part of the patch my PR does the same, but has a chance to merge upstream. The lower part I don't really understand where it comes from. @deads2k can shed light on it.

Reply:
Done. Added `UPSTREAM: 89147: aggregator: wait for complete proxy handler` prior to `UPSTREAM: <carry>: wait for apiservices on startup`